imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/TIFFImageMetadata.java

```diff
@@ -1024,6 +1024,34 @@ else if ("HorizontalPixelSize".equals(nodeName)) {
             else if ("VerticalPixelSize".equals(nodeName)) {
                 yRes = Float.parseFloat(getAttribute(child, "value"));
             }
+            else if ("ImageOrientation".equals(nodeName)) {
+                switch (getAttribute(child, "value").toLowerCase()) {
+                    case "normal":
+                        orientation = TIFFBaseline.ORIENTATION_TOPLEFT;
+                        break;
+                    case "fliph":
+                        orientation = TIFFExtension.ORIENTATION_TOPRIGHT;
+                        break;
+                    case "rotate180":
+                        orientation = TIFFExtension.ORIENTATION_BOTRIGHT;
+                        break;
+                    case "flipv":
+                        orientation = TIFFExtension.ORIENTATION_BOTLEFT;
+                        break;
+                    case "fliphrotate90":
+                        orientation = TIFFExtension.ORIENTATION_LEFTTOP;
+                        break;
+                    case "rotate270":
+                        orientation = TIFFExtension.ORIENTATION_RIGHTTOP;
+                        break;
+                    case "flipvrotate90":
+                        orientation = TIFFExtension.ORIENTATION_RIGHTBOT;
+                        break;
+                    case "rotate90":
+                        orientation = TIFFExtension.ORIENTATION_LEFTBOT;
+                        break;
+                }
```

Comment:
Can you extract this to a method (and use `return` in each `case`), I think that will make this more compact and easier to read.

Reply:
Done, extracted and null check :)